bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/types/BUnionType.java

```diff
@@ -180,7 +183,20 @@ public int hashCode() {
         return Objects.hash(super.hashCode(), memberTypes);
     }
 
+    public void setMemberTypes(BType[] memberTypes) {
+        this.memberTypes = Arrays.asList(memberTypes);
+        setFlagsBasedOnMembers();
+    }
+
     public boolean isNilable() {
+        if (memberTypes == null) {
```

Comment:
Why do we need this check? Does `isNilable` get called before the union type creation is completed?

Reply:
Yes, `hasFillerValue` is called on `populate` methods, resulting in this call. Having this check prevents the null pointer issue since we are setting the members to null when creation.